go-controller/pkg/ovn/kubevirt.go

```diff
@@ -110,18 +122,287 @@ func (oc *DefaultNetworkController) deleteDHCPOptions(pod *kapi.Pod) error {
 		}
 
-		if !isMigratedSourcePodStale {
-			if err := oc.deleteDHCPOptions(pod); err != nil {
-				return err
+		// adds a policy so that a migrated pods egress traffic
+		// will be routed to the local GR where it now resides
+		match := fmt.Sprintf("ip%s.src == %s", ipFamily, podAddress)
+		egressPolicy := nbdb.LogicalRouterPolicy{
+			Match:    match,
+			Action:   nbdb.LogicalRouterPolicyActionReroute,
+			Nexthops: []string{nodeGRAddress.IP.String()},
+			Priority: types.EgressLiveMigrationReroutePiority,
+			ExternalIDs: map[string]string{
+				kubevirt.OvnZoneExternalIDKey:           kubevirt.OvnLocalZone,
+				string(libovsdbops.VirtualMachineIndex): pod.Labels[kubevirtv1.VirtualMachineNameLabel],
+				string(libovsdbops.NamespaceIndex):      pod.Namespace,
+			},
+		}
+		if err := libovsdbops.CreateOrUpdateLogicalRouterPolicyWithPredicate(oc.nbClient, types.OVNClusterRouter, &egressPolicy, func(item *nbdb.LogicalRouterPolicy) bool {
+			return item.Priority == egressPolicy.Priority && item.Match == egressPolicy.Match && item.Action == egressPolicy.Action
+		}); err != nil {
+			return fmt.Errorf("failed adding point to point policy for pod %s/%s : %v", pod.Namespace, pod.Name, err)
+		}
+
+		// Add a route for reroute ingress traffic to the VM port since
+		// the subnet is alien to ovn_cluster_router
+		outputPort := types.RouterToSwitchPrefix + pod.Spec.NodeName
+		ingressRoute := nbdb.LogicalRouterStaticRoute{
+			IPPrefix:   podAddress,
+			Nexthop:    podAddress,
+			Policy:     &nbdb.LogicalRouterStaticRoutePolicyDstIP,
+			OutputPort: &outputPort,
+			ExternalIDs: map[string]string{
+				kubevirt.OvnZoneExternalIDKey:           kubevirt.OvnLocalZone,
+				string(libovsdbops.VirtualMachineIndex): pod.Labels[kubevirtv1.VirtualMachineNameLabel],
+				string(libovsdbops.NamespaceIndex):      pod.Namespace,
+			},
+		}
+		if err := libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(oc.nbClient, types.OVNClusterRouter, &ingressRoute, func(item *nbdb.LogicalRouterStaticRoute) bool {
+			matches := item.IPPrefix == ingressRoute.IPPrefix && item.Nexthop == ingressRoute.Nexthop && item.Policy != nil && *item.Policy == *ingressRoute.Policy
+			return matches
+		}); err != nil {
+			return fmt.Errorf("failed adding static route: %v", err)
+		}
+	}
+	return nil
+}
+
+// ensureRemoteZonePodAddressesToNodeRoute will add static routes when live
+// migrated pod belongs to remote zone to send traffic over transwitch switch
+// port of the node where the pod is running:
+//   - A dst-ip with live migrated pod ip as prefix and nexthop the pod's
+//     current node transit switch port.
+func (oc *DefaultNetworkController) ensureRemoteZonePodAddressesToNodeRoute(pod *kapi.Pod) error {
+	// DHCPOptions are only needed at the node is running the VM
+	// at that's the local zone node not the remote zone
+	if err := oc.deleteDHCPOptions(pod); err != nil {
+		return err
+	}
+	if err := oc.deleteRoutingForMigratedPod(pod); err != nil {
```

Comment:
Should this just be removing local zone routing, if anything?

Should we do the same in `ensureLocalZonePodAddressesToNodeRoute`?

Reply:
Will add a change to remove only routing resource that should run at local zone, since the created policy here is mean to be for remote zone it will not be removed if this method is called twice